tmt/utils.py

```diff
@@ -153,8 +153,73 @@ def unrooted(self) -> 'Path':
 # A stand-in variable for generic use.
 T = TypeVar('T')
 
-# A FMF context type, representing name/values context.
-FmfContextType = Dict[str, List[str]]
+
+class FmfContext(Dict[str, List[str]]):
+    """
+    Represents a fmf context.
+
+    See https://tmt.readthedocs.io/en/stable/spec/context.html
+    and https://fmf.readthedocs.io/en/latest/context.html.
+    """
+
+    @classmethod
+    def _normalize_command_line(cls, spec: List[str], logger: tmt.log.Logger) -> 'FmfContext':
+        """
+        Normalize command line fmf context specification.
+
+        .. code-block:: ini
+
+            distro=fedora-33 -> {'distro': ['fedora']}
+            arch=x86_64,ppc64 -> {'arch': ['x86_64', 'ppc64']}
+        """
+
+        return FmfContext({
+            key: value.split(',')
+            for key, value in environment_to_dict(variables=spec, logger=logger).items()})
+
+    @classmethod
+    def _normalize_fmf(
+            cls,
+            spec: Dict[str, Union[str, List[str]]],
+            logger: tmt.log.Logger) -> 'FmfContext':
+        """
+        Normalize fmf context specification from fmf node.
+
+        .. code-block:: yaml
+
+            context:
+              - distro: fedora-33
+              arch:
+                - x86_64
+                - ppc64
+        """
+
+        normalized: FmfContext = FmfContext()
+
+        for dimension, values in spec.items():
+            if isinstance(values, list):
+                normalized[str(dimension)] = [str(v) for v in values]
+            else:
+                normalized[str(dimension)] = [str(values)]
+
+        return normalized
+
+    @classmethod
+    def from_spec(cls, key_address: str, spec: Any, logger: tmt.log.Logger) -> 'FmfContext':
```

Comment:
A docstring is missing, this is a public method which whould be explained.

I would even thing that the docstrings fromt the "private" methods can be moved here maybe ...

Reply:
Added in 276eb601 and 9165eb03 . I kept the docstrings in private methods.